src/Type/Php/ArrayReplaceFunctionReturnTypeExtension.php

```diff
@@ -24,68 +24,57 @@ public function isFunctionSupported(FunctionReflection $functionReflection): boo
 
 	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
 	{
-		$args = $functionCall->getArgs();
+		$arrayTypes = $this->collectArrayTypes($functionCall, $scope);
 
-		if (count($args) < 1) {
+		if (count($arrayTypes) === 0) {
 			return ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
```

Comment:
I think you could return `null` here to simplify it

Reply:
oh right, I forgot. I definitely prefer the new way, thx for the reminder! adapted